File: library/src/main/java/org/incendo/interfaces/interfaces/CombinedInterfaceBuilder.java
package org.incendo.interfaces.interfaces;

import net.kyori.adventure.text.Component;
import org.incendo.interfaces.pane.CombinedPane;

public final class CombinedInterfaceBuilder extends AbstractInterfaceBuilder<CombinedPane, CombinedInterface> {

    private int rows = 0;
    private Component initialTitle = null;

    public int rows() {
        return this.rows;
    }

    public void rows(final int rows) {
        this.rows = rows;
    }

    public Component initialTitle() {
        return this.initialTitle;
    }

    public void initialTitle(final Component initialTitle) {
        this.initialTitle = initialTitle;
    }

    @Override
    public CombinedInterface build() {
        return new CombinedInterface(
            this.rows,
            this.initialTitle,
            this.closeHandlers(),
            this.transforms(),
            this.clickPreprocessors()
        );
    }
}

File: next/src/main/java/org/incendo/interfaces/next/interfaces/CombinedInterfaceBuilder.kt
package org.incendo.interfaces.next.interfaces

import net.kyori.adventure.text.Component
import org.incendo.interfaces.next.pane.CombinedPane

public class CombinedInterfaceBuilder internal constructor() :
    AbstractInterfaceBuilder<CombinedPane, CombinedInterface>() {

    public var rows: Int = 0
    public var initialTitle: Component? = null

    public override fun build(): CombinedInterface = CombinedInterface(
        rows,
        initialTitle,
        transforms,
        clickPreprocessors
    )
}
